utils/json-utils/src/main/scala/pl/touk/nussknacker/engine/json/serde/CirceJsonDeserializer.scala

```diff
@@ -20,6 +21,6 @@ class CirceJsonDeserializer(jsonSchema: Schema) {
   def deserialize(string: String): Validated[circe.Error, Any] = {
     val jsonObject = new JSONTokener(string).nextValue()
     jsonSchema.validate(jsonObject)
-    Validated.fromEither(CirceUtil.decodeJson[Json](string).map(JsonToTypedMapConverter.jsonToAny))
+    Validated.fromEither(CirceUtil.decodeJson[Json](string).map(JsonToObject.apply(_, SwaggerBasedJsonSchemaTypeDefinitionExtractor.swaggerType(jsonSchema))))
```

Comment:
the type should be computed only once for this class?

Reply:
done